src/main/kotlin/com/openlattice/hazelcast/serializers/DataExpirationStreamSerializer.kt

```diff
@@ -0,0 +1,55 @@
+package com.openlattice.hazelcast.serializers
+
+import com.hazelcast.nio.ObjectDataInput
+import com.hazelcast.nio.ObjectDataOutput
+import com.kryptnostic.rhizome.pods.hazelcast.SelfRegisteringStreamSerializer
+import com.openlattice.data.DataExpiration
+import com.openlattice.data.DeleteType
+import com.openlattice.edm.set.ExpirationBase
+import com.openlattice.hazelcast.StreamSerializerTypeIds
+import org.springframework.stereotype.Component
+
+@Component
+class DataExpirationStreamSerializer : SelfRegisteringStreamSerializer<DataExpiration> {
+
+    companion object {
+        private val expirationTypes = ExpirationBase.values()
+        private val deleteTypes = DeleteType.values()
+
+        @JvmStatic
+        fun serialize(out: ObjectDataOutput, obj: DataExpiration) {
+            out.writeLong(obj.timeToExpiration)
+            out.writeInt(obj.expirationBase.ordinal)
+            out.writeInt(obj.deleteType.ordinal)
+            OptionalStreamSerializers.serialize(out, obj.startDateProperty, UUIDStreamSerializer::serialize)
```

Comment:
Can use the Kotlin ones mentioned above here as well. Unsure if one is better but the Kotlin ones use Kotlin (duh 😆)